packages/discord.js/src/client/websocket/WebSocketManager.js

```diff
@@ -258,7 +258,9 @@ class WebSocketManager extends EventEmitter {
          * @param {number} id The shard id that disconnected
          */
         this.client.emit(Events.ShardDisconnect, { code, reason: reasonIsDeprecated, wasClean: true }, shardId);
-        this.debug(GatewayCloseCodes[code], shardId);
+        if (code in GatewayCloseCodes) {
+          this.debug(`Shard not resumable: ${GatewayCloseCodes[code]}`, shardId);
+        }
```

Comment:
My exact thoughts originally. Two things, though:
- Whereas the Discord gateway error types are descriptive, the message `Shard not resumable: Normal` is probably equally as unhelpful as `undefined`.
- Deliberate websocket closures are expected not to resume by definition. A better option for clear logging in this case might be to attach an explicit debug reason in `WebSocketManager#destroy`:
  ```diff
      /**
       * Destroys this manager and all its shards.
       * @private
       */
      async destroy() {
        if (this.destroyed) return;
        // TODO: Make a util for getting a stack
        this.debug(`Manager was destroyed. Called by:\n${new Error().stack}`);
        this.destroyed = true;
  -     await this._ws?.destroy({ code: CloseCodes.Normal });
  +     await this._ws?.destroy({ code: CloseCodes.Normal, reason: 'Manager was destroyed' });
      }
  ```

Reply:
I'm with Almeida here.

> Whereas the Discord gateway error types are descriptive, the message Shard not resumable: Normal is probably equally as unhelpful as undefined.

While probably "unhelpful" for the average user, it is useful for me as a maintainer. As per the "normal" thing, the `CloseCodes` could just not be resolved against the strings, considering they're mostly for "internal" use (e.g. our `Resuming` code just being whatever 4xxx number I decided on at the time) - `1000` isn't a whole lot better for the average person, but anyone experienced with the WS protocol will know what that means.

> Deliberate websocket closures are expected not to resume by definition. A better option for clear logging in this case might be to attach an explicit debug reason in `WebSocketManager#destroy`

Sure, this could also work. Either way, I'm unhappy with the current state of this PR.